docs/src/tutorial.md

```diff
@@ -139,3 +139,172 @@ sol = WorldDynamics.solve(system, (1900, 2100))
+```
+Note that the function `gettables` is exactly the one that has been used above while updating a model with modern data.
+
+We can now define a scenario by simply invoking the function `non_renewable_stock` and by returning the ODE system returned by this function. This is done in the file `scenarios.jl` (we assume that also this file is contained in the directory `nonrenewablestock`) which contains the following code.
+
+```
+function nrs_run(; kwargs...)
+    @named nrs = non_renewable_stock(; kwargs...)
+    return nrs
+end
+```
+Observe that if the model contains multiple systems and/or multiple subsystems, then the ODE systems returned by all the subsystems have to be composed by using the function `compose` (which also asks for the connections between the variables declared and used in different subsystems). An example of a scenario using multiple systems and subsystems is defined in the file `scenarios.jl` included in the directory `world2` within the `World` model directory.
+
+Finally, the model can be solved and simulated by using the `solve` and the `plotvariables` functions that we already used above. In particular, the file `plots.jl` (we assume that also this file is contained in the directory `nonrenewablestock`) does it in order to reproduce Figure 3.9 of the chapter of the above mentioned book.
+
+```
+using ModelingToolkit
+using DifferentialEquations
+
+function nrs_run_solution()
+    isdefined(@__MODULE__, :_solution_nrs_run) && return _solution_nrs_run
+    global _solution_nrs_run = solve(nrs_run(), (0, 200), solver=Tsit5(), dt=0.015625, dtmax=0.015625)
+    return _solution_nrs_run
+end
+function _variables_nrs()
+    @named nrs = non_renewable_stock()
+    variables = [
+        (nrs.capital, 0, 30, "Capital"),
+        (nrs.extraction, 0, 15, "Extraction"),
+        (nrs.investment, 0, 2, "Investment"),
+        (nrs.depreciation, 0, 2, "Investment"),
+        (nrs.resource, 0, 1000, "Resource"),
+    ]
+    return variables
+end
+
+fig_3_9(; kwargs...) = plotvariables(nrs_run_solution(), (t, 0, 200), _variables_nrs(); title="Simulation output showing stocks and flows", showaxis=false, showlegend=true, kwargs...)
+```
+Note that, for performance reasons, the definition of the function `nrs_run_solution` starts by checking whether the solution of the model is already available: in this case, nothing is done.
+
+### Integrating the model into WorldDynamics
+
+In order to integrate the new model in the `WorldDynamics` framework, we can now define a Julia module `Duggan.jl` as follows (we assume that this source file is contained in the directory `Duggan`).
+
+```
+module Duggan
+using ModelingToolkit
+using WorldDynamics
+include("NonRenewableStock.jl")
+end
+```
+The file `NonRenewableStock.jl` (we assume that also this file is contained in the directory `Duggan`) simply includes all the Julia source files we have written above.
+
+```
+module NonRenewableStock
+using WorldDynamics
+using ModelingToolkit
```

Comment:
This may be a very naive question, since we are currently doing that in the models we implemented, but... do we need to re-include these two modules, which are already included in `Duggan.jl`.

Reply:
Don't know...